### PR TITLE
Make resources in agent and operator helm chart configurable

### PR DESCRIPTION
**Pull request by maxbischoff**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Enables users of the  helm chart to provide resource requests and limits to the containers of the agent and operator deployment/daemonset. Also provides default requests for the agent's initContainer.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10296)

### CODE REVIEW

File: install/kubernetes/cilium/charts/agent/values.yaml [hunk 1]
```diff
@@ -8,3 +8,14 @@ maxUnavailable: 2
...
   enabled: false
   eventTypes: []
+  # Specifies the resources for the monitor sidecar container if activated
+  resources: {}
+
+# Specifies the resources for the agent container
+resources: {}
+
+# Specifies the resources for the clean-cilium-state init container
+initResources:
```
Comment (maxbischoff) [inline install/kubernetes/cilium/charts/agent/values.yaml:18]: I am not particularly fond of this naming but also could not come up with anything better. If you have a suggestion I would happily change it.

**Comment by joestringer** [comment]
test-me-please

File: install/kubernetes/cilium/charts/agent/values.yaml [hunk 2]
```diff
@@ -8,3 +8,14 @@ maxUnavailable: 2
...
+  requests:
+    cpu: "100m"
+    memory: "100Mi"
```
Comment (aanm) [inline install/kubernetes/cilium/charts/agent/values.yaml:21]: Having these requests with default values is hard has it highly depends on the cluster size.

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/agent/values.yaml (hunk 2 above)
Comment (aanm) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: cc @tklauser @tgraf

Comment (joestringer) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: Cleaning up local node cilium state depends on cluster size? I would think it's the same set of BPF maps and so on regardless of cluster size. I can understand that the node size could play in, in that we have a number of BPF templates and progs depending on the number of local pods. Are there other aspects you're concerned with?

Comment (tklauser) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: If I understand https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource/ correctly, this is not limiting the memory use but only specifies the requested memory in order to influence scheduling of the pod. IMO `100Mi` memory seems fairly low for a standard Cilium pod.

@maxbischoff how did you determine these numbers?

Comment (maxbischoff) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: This memory request is only for the init container of the cilium pod, I left the requests for the other containers open to the user on purpose.
We just started using cilium so I cannot tell if having `100Mi` for an initContainer is too little, but with other network plugins (calico) that was plenty in our experience.
I can also remove this default if you would prefer so.

In my opinion, suggesting values for the cilium agent containers is a separate issue from the one I am trying to solve (making the resources configurable)

Comment (aanm) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: Ah this is for the init container, then I would say this is plenty enough. And like @tklauser said this is the requested memory not the limit so these limits look sane to me

**Review by aanm — APPROVED** [review]

File: install/kubernetes/cilium/charts/agent/values.yaml (hunk 2 above)
Comment (tklauser) [reply install/kubernetes/cilium/charts/agent/values.yaml:21]: Thanks for elaborating. I didn't get that this is only for the init container. In that case LGTM as well.

**Review by tklauser — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Pull request merged**